client/src/components/collection/sendCollectionToNotebook.vue

```diff
@@ -32,50 +34,96 @@ export default {
+      console.log("url=" + url)
       window.open(url);
     },
 
     nbBinderUrl ( NbConfig,  collectionObj, page=''){
-      let collection = JSON.stringify(collectionObj)
-      collection  = collection.replace('#',"<hash>")
-      collection = encodeURIComponent(collection)
-
-
-
-      //let binderBase = "https://mybinder.org/v2/gh/earthcube/NotebookTemplates.git/geocodes_template"
-
-      //let nbParams = `?contenturl=${contentUrl}&format=${format}&urn=${urn}`
+		let coQ = collectionObj.queries
+		console.log("coQ")
+		console.log(coQ)
+		let coT = collectionObj.tools
+		let coD = collectionObj.datasets
+		let URNq = coQ.map(x => x.name)
+		console.log(URNq)
+		let URNt = coT.map(x => x.g)
+		let URNd = coD.map(x => x.g)
+		let des = collectionObj.description
+		let name = des.name
+		var URNdict = {
+		name: name,
+		datasets: URNd,
+		queries: URNq,
+		tools: URNt
+		};
+		console.log("URNd_t_q")
+		console.log(URNd)
+		console.log(URNt)
+		console.log(URNq)
+		console.log("URNdict")
+		console.log(URNdict)
+		let collection = JSON.stringify(URNdict)
+		console.log("collection")
+		console.log(collection)
+		//collection  = collection.replace('#',"<hash>")
+		collection = encodeURIComponent(collection)
+		console.log(collection)
+
+		//then binder template will need:
+		//ds = ipyparams.params['collection']
+		//print(ds)
+		//dso = json.loads(ds)
+		//# if this cell fails the first run.
+		//# run a second time, and it works.
+		//d=dso.get('datasets')
+		//t=dso.get('tools')
+		//q=dso.get('queries')
+		//print(f'd={d} t={t} q={q}')
+		//binder has own sparql.ipynb now, &colab's is changing as well
+		//hard coded to a machine that I can update more easily till collections more stable
+
+
+
+		//let binderBase = "https://mybinder.org/v2/gh/earthcube/NotebookTemplates.git/geocodes_template"
+
+		//let nbParams = `?contenturl=${contentUrl}&format=${format}&urn=${urn}`
+		//let nbParams = `?contenturl=urn=${urn}`  //try?
       //const binderBaseT = _.template(NbConfig.baseurl, FacetsConfig.ES_TEMPLATE_OPTIONS)
       //const dispatcherPageT = _.template(NbConfig.dispatcherPage, FacetsConfig.ES_TEMPLATE_OPTIONS)
-      const serverBase = NbConfig.baseurl
-      let dispatcherPage =NbConfig.dispatcherPage
+
+      //const serverBase = NbConfig.baseurl
+      const serverBase = "http://141.142.218.86:3031/mkC/"
+
+//      let dispatcherPage =NbConfig.dispatcherPage
 
       const nbParamsT =_.template(NbConfig.contentQuery, FacetsConfig.ES_TEMPLATE_OPTIONS)
       const nbPageT =_.template(NbConfig.pageTemplate, FacetsConfig.ES_TEMPLATE_OPTIONS)
 
       let nbParams = nbParamsT({collection:collection})
       let nbPage = nbPageT({notebooktorun:page})
       let params = ''
-      if (Object.prototype.hasOwnProperty.call(NbConfig,'dispatcherPage')) {
-        params = `${dispatcherPage}&${nbParams}&${nbPage}`
-      } else {
+//     if (Object.prototype.hasOwnProperty.call(NbConfig,'dispatcherPage')) {
```

Comment:
Again, this is handles as a configuration parameter. empty nbconfig.dispatcherpage,  then params = `${nbParams}&${nbPage}`